File: java/src/main/java/org/benchmark/polymorphism/Cart1.java
package org.benchmark.polymorphism;

public class Cart1 {
    private Book p;
    private int qty;
    public Cart1(Book b) {
        this.p = b;
        this.qty = 10;
    }
    public int total() { return p.price() * 2; }
    public Cart2 with(Movie m) {
        Cart2 c2 = new Cart2(m);
        return c2;
    }
}

File: cpp/Examples/Cart1.cpp
#include "Cart1.h"

Cart1::Cart1(Book *p) {
    this->p = p;
    this->qty = 10;
}

int Cart1::total() {
    return p->price() * qty;
}

Cart2 Cart1::with(Movie *m) {
    Cart2 c2 = Cart2(m);
    return c2;
}
